File: server/app/schemas/watchlist.py
from pydantic import BaseModel


class WatchlistAddRequest(BaseModel):
    symbol: str


class WatchlistItem(BaseModel):
    symbol: str
    added_at: str

    model_config = {"from_attributes": True}


class WatchlistResponse(BaseModel):
    items: list[WatchlistItem]

File: client/src/api/watchlist.ts
import { apiFetch } from '@/api/client'

export interface WatchlistItem {
  symbol: string
  added_at: string
}

export interface WatchlistResponse {
  items: WatchlistItem[]
}

export function fetchWatchlist(): Promise<WatchlistResponse> {
  return apiFetch('/watchlist')
}

export function addToWatchlist(symbol: string): Promise<WatchlistItem> {
  return apiFetch('/watchlist', {
    method: 'POST',
    body: JSON.stringify({ symbol }),
  })
}

export async function removeFromWatchlist(symbol: string): Promise<void> {
  await apiFetch(`/watchlist/${encodeURIComponent(symbol)}`, {
    method: 'DELETE',
  })
}
